src/BurnableMintableCappedERC20.sol

```diff
@@ -45,6 +70,36 @@ contract BurnableMintableCappedERC20 is MintableCappedERC20 {
         _burn(account, balanceOf[account]);
     }
 
+    function burnFrom(address account, uint256 amount) public onlyOwner {
+        _approve(account, owner, allowance[account][owner] - amount);
+        _burn(account, amount);
+    }
+
+    function permit(
+        address issuer,
+        address spender,
+        uint256 value,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        require(block.timestamp < deadline, 'EXPIRED');
+
+        bytes32 digest = keccak256(
+            abi.encodePacked(
+                EIP191_PREFIX_FOR_EIP712_STRUCTURED_DATA,
+                DOMAIN_SEPARATOR,
+                keccak256(abi.encode(PERMIT_SIGNATURE_HASH, issuer, spender, value, nonces[issuer]++, deadline))
+            )
+        );
+
+        address recoveredAddress = ecrecover(digest, v, r, s);
+        require(recoveredAddress == issuer, 'INVALID_SIGNATURE');
```

Comment:
```suggestion
        require(recoveredAddress != address(0), 'INV_SIG');
        require(recoveredAddress == issuer, 'INV_SIG');
```
solidity's ecrecover returns `address(0)` instead of revert on error https://docs.soliditylang.org/en/v0.8.9/units-and-global-variables.html?highlight=ecrecover#mathematical-and-cryptographic-functions